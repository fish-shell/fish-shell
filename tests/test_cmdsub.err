# Command sub at the limit should fail.
fish: Too much data emitted by command substitution so it was discarded

set b (string repeat -n 512 x)
      ^
# Command sub over the limit should fail.
fish: Too much data emitted by command substitution so it was discarded

set -l x (string repeat -n $argv x)
         ^
in function 'subme'
	called on standard input
	with parameter list '513'

in command substitution
	called on standard input

# Same builtin in a command substitution is affected
fish: Too much data emitted by command substitution so it was discarded

echo this will fail (string repeat --max 513 b) to output anything
                    ^
