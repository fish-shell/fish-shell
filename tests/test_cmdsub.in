# This tests various corner cases involving command substitution. Most
# importantly the limits on the amount of data we'll substitute.

set FISH_READ_BYTE_LIMIT 512

function subme
    set -l x (string repeat -n $argv x)
    echo $x
end

echo '# Command sub just under the limit should succeed.'
set a (subme 511)
show a

echo '# Command sub at the limit should fail.'
echo '# Command sub at the limit should fail.' >&2
set b (string repeat -n 512 x)
set saved_status $status
test $saved_status -eq 122
or echo expected status 122, saw $saved_status >&2
show b

echo '# Command sub over the limit should fail.'
echo '# Command sub over the limit should fail.' >&2
set c (subme 513)
show c

echo '# Make sure output from builtins outside of command substitution is not affected'
string repeat --max 513 a

echo '# Same builtin in a command substitution is affected' >&2
echo this will fail (string repeat --max 513 b) to output anything
set saved_status $status
test $saved_status -eq 122
or echo expected status 122, saw $saved_status >&2
