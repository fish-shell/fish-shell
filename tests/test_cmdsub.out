# Command sub just under the limit should succeed.
$a count=1
$a[1]=|xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx|
# Command sub at the limit should fail.
$b is not set
# Command sub over the limit should fail.
$c count=1
$c[1]=||
# Make sure output from builtins outside of command substitution is not affected
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
